modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -3254,19 +3271,73 @@ public LocalSnapshotSender(String snpName, @Nullable String snpPath) {
 
                 assert totalBytes % pageSize == 0 : "Given file with delta pages has incorrect size: " + fileIo.size();
 
+                int pagesCnt = (int)(totalBytes / pageSize);
+
+                Integer batchSize = deltaSortBatch.get();
+
+                GridIntIterator iter;
+
+                if (batchSize == null || batchSize <= 0)
+                    iter = U.forRange(0, pagesCnt);
+                else {
+                    iter = new GridIntIterator() {
+                        private int idx = 0;
+
+                        private Iterator<Integer> sortedIter;
+
+                        @Override public boolean hasNext() {
+                            if (sortedIter == null || !sortedIter.hasNext()) {
+                                try {
+                                    advance();
+                                }
+                                catch (Exception e) {
+                                    throw new IgniteException(e);
+                                }
+                            }
+
+                            return sortedIter.hasNext();
+                        }
+
+                        @Override public int next() {
+                            return sortedIter.next();
+                        }
+
+                        private void advance() throws Exception {
+                            TreeMap<Integer, Integer> sorted = new TreeMap<>();
+
+                            while (idx < pagesCnt && sorted.size() < batchSize) {
+                                transferRateLimiter.acquire(pageSize);
```

Comment:
We are limiting only writes, why we should use the same for reads?

Reply:
I assume this is a total throttling limit to a disk to reduce snapshot process affect.